src/GafferDispatch/Dispatcher.cpp

```diff
@@ -434,16 +438,13 @@ class Dispatcher::Batcher
 			{
 				float frame = task.context()->getFrame();
 				std::vector<float> &frames = batch->frames();
-				if( std::find( frames.begin(), frames.end(), frame ) == frames.end() )
```

Comment:
I've removed this check because I believe it's already taken care of by the fact that we'd have [exited from this method early](https://github.com/GafferHQ/gaffer/pull/2064/commits/8fa14427133da7b0e3dd5f648413b01ab24461cb#diff-00530c3fa5d5313e32703b777a4bde56R401) if the frame had been processed already, and the the code for checking batch size limits has always assumed that a frame will be added unconditionally anyway. It's also a potential performance hazard for large batch sizes (or long sequence executions). Hopefully this makes sense, but you might know something I don't Andrew?